velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -384,6 +406,206 @@ TEST_F(Re2FunctionsTest, likePattern) {
+  EXPECT_EQ(
+      determinePatternKind("%_%%"),
+      std::make_pair(PatternKind::kWildcardTwo, 1));
+  EXPECT_EQ(
+      determinePatternKind("_b%%__"), std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%_%p"), std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("presto"), std::make_pair(PatternKind::kFixed, 6));
+  EXPECT_EQ(
+      determinePatternKind("hello"), std::make_pair(PatternKind::kFixed, 5));
+  EXPECT_EQ(determinePatternKind("a"), std::make_pair(PatternKind::kFixed, 1));
+  EXPECT_EQ(
+      determinePatternKind("helloPrestoWorld"),
+      std::make_pair(PatternKind::kFixed, 16));
+  EXPECT_EQ(
+      determinePatternKind("aBcD_"), std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%aBc_D%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("presto%"), std::make_pair(PatternKind::kPrefix, 6));
+  EXPECT_EQ(
+      determinePatternKind("hello%%"), std::make_pair(PatternKind::kPrefix, 5));
+  EXPECT_EQ(
+      determinePatternKind("a%"), std::make_pair(PatternKind::kPrefix, 1));
+  EXPECT_EQ(
+      determinePatternKind("helloPrestoWorld%%%"),
+      std::make_pair(PatternKind::kPrefix, 16));
+  EXPECT_EQ(
+      determinePatternKind("aBcD%%e%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("aBc_D%%"),
+      std::make_pair(PatternKind::kGeneric, 0));
+
+  EXPECT_EQ(
+      determinePatternKind("%presto"), std::make_pair(PatternKind::kSuffix, 6));
+  EXPECT_EQ(
+      determinePatternKind("%%hello"), std::make_pair(PatternKind::kSuffix, 5));
+  EXPECT_EQ(
+      determinePatternKind("%a"), std::make_pair(PatternKind::kSuffix, 1));
+  EXPECT_EQ(
+      determinePatternKind("%%%helloPrestoWorld"),
+      std::make_pair(PatternKind::kSuffix, 16));
+  EXPECT_EQ(
+      determinePatternKind("%%_%aBcD"),
+      std::make_pair(PatternKind::kGeneric, 0));
+  EXPECT_EQ(
+      determinePatternKind("%%a%%BcD"),
+      std::make_pair(PatternKind::kGeneric, 0));
+}
+
+TEST_F(Re2FunctionsTest, likePatternWildcard) {
+  auto like = [&](std::optional<std::string> str,
+                  std::optional<std::string> pattern) {
+    return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+  };
+
+  EXPECT_EQ(like("", ""), true);
+  EXPECT_EQ(like("", "%"), true);
+  EXPECT_EQ(like("abc", "%%%%"), true);
+  EXPECT_EQ(like("abc", "%%"), true);
+  EXPECT_EQ(like("abc", "%_%_%"), true);
+  EXPECT_EQ(like("abc", "_%_%_%"), true);
+  EXPECT_EQ(like("a", "_"), true);
+  EXPECT_EQ(like("ab", "__"), true);
+  EXPECT_EQ(like("abc", "___"), true);
+  EXPECT_EQ(like("", "_"), false);
+  EXPECT_EQ(like("ab", "_"), false);
+  EXPECT_EQ(like("abcd", "___"), false);
+  EXPECT_EQ(like("acb", "%ab_%"), false);
+  EXPECT_EQ(like("abcd", "_abc%d"), false);
+  EXPECT_EQ(like("abcd", "%ab_c%"), false);
+
+  std::string input = generateRandomString(alphaNumCharSet_);
```

Comment:
It is not a good practice to name variables a1, a2, a3. Consider refactoring to shorten generateRandomString to generateString and inline inputXxx variables.

```
EXPECT_TRUE(like(generateString(kAlphaNumCharSet), generateString(kWildcardCharSet)))
```

Reply:
Removed these tests since the random string generation is removed as well, thank you!